e2e/fiat_tf_test.go

```diff
@@ -0,0 +1,529 @@
+
+}
+
+func TestFiatTFConfigureMinter(t *testing.T) {
+	if testing.Short() {
+		t.Skip()
+	}
+	t.Parallel()
+
+	ctx := context.Background()
+
+	nw := nobleSpinUp(t, ctx, true)
+	noble := nw.chain
+	val := noble.Validators[0]
+
+	// ACTION: Configure minter while TF is paused
+	// EXPECTED: Request fails; Minter is not configured
+
+	// configure new minter controller and minter
+	w := interchaintest.GetAndFundTestUsers(t, ctx, "default", math.OneInt(), noble, noble)
+	minterController1 := w[0]
+	minter1 := w[1]
+
+	_, err := val.ExecTx(ctx, nw.fiatTfRoles.MasterMinter.KeyName(), "fiat-tokenfactory", "configure-minter-controller", minterController1.FormattedAddress(), minter1.FormattedAddress())
+	require.NoError(t, err, "error configuring minter controller")
+
+	showMCRes, err := showMinterController(ctx, val, minterController1)
+	require.NoError(t, err, "failed to query show-minter-controller")
+	expectedShowMinterController := fiattokenfactorytypes.QueryGetMinterControllerResponse{
+		MinterController: fiattokenfactorytypes.MinterController{
+			Minter:     minter1.FormattedAddress(),
+			Controller: minterController1.FormattedAddress(),
+		},
+	}
+
+	require.Equal(t, expectedShowMinterController.MinterController, showMCRes.MinterController)
+
+	pauseFiatTF(t, ctx, val, nw.fiatTfRoles.Pauser)
+
+	allowance := int64(10)
+
+	_, err = val.ExecTx(ctx, minterController1.KeyName(), "fiat-tokenfactory", "configure-minter", minter1.FormattedAddress(), fmt.Sprintf("%duusdc", allowance))
+	require.ErrorContains(t, err, "minting is paused")
+
+	_, err = showMinters(ctx, val, minter1)
+	require.Error(t, err, "minter found; configuring minter should not have succeeded")
+
+	unpauseFiatTF(t, ctx, val, nw.fiatTfRoles.Pauser)
+
+	// ACTION: Configure minter from a minter controller not associated with the minter
+	// EXPECTED: Request fails; Minter is not configured with new Minter Controller but old Minter retains its allowance
+	// Status:
+	// minterController1 -> minter1 (un-configured)
+	// gw minterController -> gw minter
+
+	// reconfigure minter to ensure balance does not change
+	configureMinter(t, ctx, val, minterController1, minter1, allowance)
+
+	differentAllowance := allowance + 99
+	_, err = val.ExecTx(ctx, nw.fiatTfRoles.MinterController.KeyName(), "fiat-tokenfactory", "configure-minter", minter1.FormattedAddress(), fmt.Sprintf("%duusdc", differentAllowance))
+	require.ErrorContains(t, err, "minter address ≠ minter controller's minter address")
+
+	showMintersRes, err := showMinters(ctx, val, minter1)
+	require.NoError(t, err, "failed to query show-minters")
+	expectedShowMinters := fiattokenfactorytypes.QueryGetMintersResponse{
+		Minters: fiattokenfactorytypes.Minters{
+			Address: minter1.FormattedAddress(),
+			Allowance: sdktypes.Coin{
+				Denom:  "uusdc",
+				Amount: math.NewInt(allowance),
+			},
+		},
+	}
+	require.Equal(t, expectedShowMinters.Minters, showMintersRes.Minters, "configured minter allowance is not as expected")
```

Comment:
I'm not quite following this case. I think it's covering the case where a minter controller attempts to configure a minter it's not associated with? Is there possibly a clearer way of representing. I'm imagining future me without the E2E test spec sheet in front of me having a bad time here.

Reply:
You are correct. I added another comment in [4299081](https://github.com/noble-assets/noble/pull/368/commits/4299081192ca3f955b72c80f0059a9358c58ae57) to hopefully clarify further.
If you have an idea on a better way to approach this one, lmk!